src/query/config/src/outer_v0.rs

```diff
@@ -464,6 +483,7 @@ impl HiveCatalogConfig {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(default)]
 pub struct CacheConfig {
     #[serde(rename = "type")]
     pub cache_type: String,
```

Comment:
We can try the enum instead of the string?
```
#[derive(clap::ArgEnum, Debug, Clone, Serialize, Deserialize)]
pub enum CacheType {
    Fs,
    Moka,
    Redis,
}
```

Reply:
We does use an enum in our inner config. For outer config, we use the simplest representation instead (to avoid play with serde and clap.)